### ISSUE TITLE
Oboformat reader/writer uses old oboparser, not the one in the current owlapi

### ISSUE THREAD

**Issue by cmungall** [protegeproject/protege#151]
Protege 5 beta16 seems to use owlapi3.5.0, which means that obo saves are messed up.

I load this ontology:

```
ontology: t

[Term]
id: X:1
name: x1
is_a: X:2

[Term]
id: X:2
name: x2
```

I edited the name/label, then saved.

I get this:
```
format-version: 1.2
date: 12:03:2015 17:36
saved-by: cjm
auto-generated-by: The OWL API (version 3.5.1)
id_space: t http://purl.obolibrary.org/obo/t.owl#
id_space: rdfs http://www.w3.org/2000/01/rdf-schema#
id_space: obo http://purl.obolibrary.org/obo/
id_space: owl http://www.w3.org/2002/07/owl#
id_space: oboInOwl http://www.geneontology.org/formats/oboInOwl#
id_space: xsd http://www.w3.org/2001/XMLSchema#
id_space: rdf http://www.w3.org/1999/02/22-rdf-syntax-ns#


! ----------------------  CLASSES  -------------------------

[Term]
id: X_1
name: x1
namespace: obo
is_a: X_2
id: X:1

[Term]
id: X_2
name: x2 zzz
namespace: obo
is_a: Thing
id: X:2


! ----------------------  PROPERTIES  -------------------------


! ----------------------  INSTANCES  -------------------------
```
this looks like the ancient obo writer. The header says generated by owlapi 3.5.1, which I believe should have the real oboformat writer, so this is puzzling.

cc @sesuncedu @dosumis

**Comment by sesuncedu** [comment]
It's not using the renderer you think it's using - which ever one it is you
think it's using, it's not that one.  Or if it is, it's using something
else for Exceptions.
Also different repository.

If that made sense, I must have left something else out.  3.5.2 should work
for this.

https://github.com/protegeproject/protege-owlapi/blob/master/src/main/java/org/protege/owlapi/apibinding/ProtegeOWLManager.java#L27

**Comment by sesuncedu** [comment]
* some changes I have made are in different bits of protege (haven't done a
pull request for e.g. the ui format error handler class).

  Also, protege likes to build against  different versions of the owlapi to
the one that is deployed, and the api change for the obo classes makes OSGI
even less helpful that usual.

There is a Big Push to get some of the key plugins ready to move protege on
to OWLAPI version 4 .
It'll all be over by <Upcoming Holiday Season>

**Comment by cmungall** [comment]
https://github.com/protegeproject/protege-owlapi/blob/master/src/main/java/org/protege/owlapi/apibinding/ProtegeOWLManager.java#L27

It should be using this, shouldn't it? org.semanticweb.owlapi.oboformat.OBOFormatRenderer

**Comment by sesuncedu** [comment]
There's an ugly hack in the osgi module in  3.5.2 that creates stub classes
that point at the right classes (theres's a subclass of the new class with
the same name as the old class that was added back to deal with this.

These caused other things to get confused. At one point the  owlapi version
4 snapshot thought it was version 3 (I was making too many fixes on master,
and ignazio forward ported one line too many :)

**Comment by sesuncedu** [comment]
Pull request merged to protege-owlapi, which needs a release - then the dependency in org.protege.editor.owl can  beadjusted to point to the new version. 
https://github.com/protegeproject/protege-owlapi/pull/11

**Comment by cmungall** [comment]
Thanks! Looks like this was merged 21 days ago. Protege 5 beta17 was released 17 days ago, but doesn't seem to include this? Save in obo is still broken.

**Comment by cmungall** [comment]
Any progress on this? We'd really like to use Protege in projects where the source is .obo

**Comment by sesuncedu** [comment]
I think this was fixed in 3.5.1 ; if not it was fixed in 3.5.2 (with some
horribly hacky classes that I  added in to the osgi distribution so that
some magic specific class references in Protégé would be satisfied,  with
delegation to the new parser).

((Also,  did you get a chance to look at the oboformat model memory tuning
that I did on Saturday to help deal with the problem of the parser building
a complete obo document in memory before converting to OWL? This was
causing problems with (amongst others) GAZ and CHEBI-  GAZ used over 1GB
even before the conversion process started. These changes have not yet been
back or forward ported to version 3 or 5, in case they are bad for other
OBO contexts.))

Of course, swing is going to want to double up the amount of memory used
once protégé is given the parsed obo document, due to the horrible way it
handles lists and trees.

On Mon, Aug 10, 2015, 11:50 AM Chris Mungall <notifications@github.com>
wrote:

Any progress on this? We'd really like to use Protege in projects where the
source is .obo

**Comment by dosumis** [comment]
Hi all,

This is a big blocker for GO.  Any chance of progress on this?

Cheers,
David

**Comment by sesuncedu** [comment]
Matthew Horridge has been updating various protégé owlapi dependencies
recently, so I would guess that there may be a new beta coming out soon.

If I recall, it ought to be possible to drop an updated
owlapi-osgidistribution into the plugins folder ; there's some code in
Protégé itself that explicitly adds the old parser by name - the kludge is
supposed to work around that.

**Comment by tudorache** [comment]
A new beta is coming up soon, likely at the very beginning of October. The new beta 18 is using OWL API 3.5.3 and will have the following output (after the rename):

format-version: 1.2
ontology: t

[Term]
id: X:1
name: x_1
is_a: X:2 ! x_2

[Term]
id: X:2
name: x_2

Is this the expected format?

The Protege beta 18 snapshot can already be built from github for further testing.

@matthewhorridge can add more details, if needed.

**Comment by matthewhorridge** [comment]
FYI, you can download a SNAPSHOT of the beta-18 from these links:

[5.0.0-beta-18-SNAPSHOT OS X](https://oss.sonatype.org/content/repositories/snapshots/edu/stanford/protege/protege-distribution/5.0.0-beta-18-SNAPSHOT/protege-distribution-5.0.0-beta-18-20150911.221122-1-os-x.zip)

[5.0.0-beta-18-SNAPSHOT Linux](https://oss.sonatype.org/content/repositories/snapshots/edu/stanford/protege/protege-distribution/5.0.0-beta-18-SNAPSHOT/protege-distribution-5.0.0-beta-18-20150911.221122-1-linux.tar.gz)

[5.0.0-beta-18-SNAPSHOT Platform Independent](https://oss.sonatype.org/content/repositories/snapshots/edu/stanford/protege/protege-distribution/5.0.0-beta-18-SNAPSHOT/protege-distribution-5.0.0-beta-18-20150911.221122-1-platform-independent.zip)

[5.0.0-beta-18-SNAPSHOT Windows](https://oss.sonatype.org/content/repositories/snapshots/edu/stanford/protege/protege-distribution/5.0.0-beta-18-SNAPSHOT/protege-distribution-5.0.0-beta-18-20150911.221122-1-win.zip)

**Comment by cmungall** [comment]
I can confirm that this works as expected. Many thanks! Have a great weekend!

**Comment by cmungall** [comment]
And chance of a beta 18 release?

**Comment by cooperl09** [comment]
Any news of a Protege 5 beta 18 update?  We are using the snapshot version from above and having ongoing issues with instability. Frequently locks up and or crashes.  Also, often on saving it will write a blank file and have to be reset to last commit.

**Comment by matthewhorridge** [comment]
A beta-19 release is imminent (like in the next day or two).

Cheers,

Matthew

On 15 Dec 2015, at 11:53, Laurel Cooper <notifications@github.com<mailto:notifications@github.com>> wrote:


Any news of a Protege 5 beta 18 update? We are using the snapshot version from above and having ongoing issues with instability. Frequently locks up and or crashes. Also, often on saving it will write a blank file and have to be reset to last commit.

**Comment by matthewhorridge** [comment]
@cooperl09 what version of Java are you using?

**Comment by cooperl09** [comment]
Looks like version 8 build 66

**Comment by matthewhorridge** [comment]
Is it the Oracle JDK or the Open JDK?

**Comment by cooperl09** [comment]
I think it is the Oracle, judging by this popup.  
![screen shot 2015-12-15 at 12 58 49 pm](https://cloud.githubusercontent.com/assets/7378000/11823929/b60b62b2-a32b-11e5-8ac8-6517d00b684d.png)

**Comment by tudorache** [comment]
@cooperl09 Thank you! What kind of instability? Any kind of details would be very helpful.

**Comment by cooperl09** [comment]
It frequently locks up and or crashes. Also, often on saving it will write a blank file and have to be reset to last commit.
When this happens you get a long Java error message.  I will try to capture it next time.

**Comment by tudorache** [comment]
Does the screen become scrambled, maybe similar to this #226, or not? When it "locks", it means that the user interface becomes not responsive, and it is not even possible to save (e.g., with the menu, or Ctrl+S)? 

If you have a set of steps that reproduces the behavior, then we could debug it. The logs or console output would be very useful, too. Thank you!

**Comment by cooperl09** [comment]
Yes, sometimes it looks like the image in #226. Other times it will lock up as described above.  
Often on saving it will write a blank file and have to be reset to last commit.
When this happens you get a long Java error message
Including my colleague @austinmeier on the discussion, as he is using it right now as well.

**Comment by austinmeier** [comment]
I have been seeing some different behavior:  after moving an entity to a new location in the class hierarchy tab, the UI becomes unresponsive.  I can click other objects, and the annotations will display, but I cannot edit them or add new ones.  In addition, they are not highlighted within the class hierarchy window either.   The Description changes to the proper entity name, but the equivalent axioms remain from the previously selected term.   I hope that makes sense.

This error shows repeatedly in my terminal:  
Error logged
java.lang.ArrayIndexOutOfBoundsException
Uncaught Exception in thread AWT-EventQueue-0
java.lang.ArrayIndexOutOfBoundsException


My system: 
 Mac OS version 10.10.5 (14F1021)
java version "1.6.0_65"
Java(TM) SE Runtime Environment (build 1.6.0_65-b14-468-11M4828a)
Java HotSpot(TM) 64-Bit Server VM (build 20.65-b04-468, mixed mode)

**Comment by tudorache** [comment]
Thank you, this is very useful! 
This is bug #226. We just found the problem with that one and it is caused by a Java bug. We are currently looking for a workaround.

**Comment by sesuncedu** [comment]
Earliest back-ported fix is in  1.8.0_60b23
<https://bugs.openjdk.java.net/browse/JDK-8081054?focusedCommentId=13822689&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-13822689>

**Comment by matthewhorridge** [comment]
Was fixed in beta-18 - see above comments.
